nix/module.nix

```diff
@@ -30,7 +31,20 @@ in
 
   config = mkIf cfg.enable {
     hardware.tuxedo-keyboard.enable = true;
-    boot.kernelModules = [ "tuxedo_io" ];
+    boot.kernelModules = [
+      # Tuxedo Control Center has a requirement on the minimum version
+      # of "tuxedo_io" kernel module.
+      # The exact requirement is currently in the 
+      # "src/native-lib/tuxedo_io_lib/tuxedo_io_ioctl.h" file of tuxedo-control-center
+      # (i.e. the #define of MOD_API_MIN_VERSION).
+      # The respective version of the module itself is in the 
+      # "src/tuxedo_io/tuxedo_io.c" file of tuxedo-keyboard
+      # (i.e. the #define of MODULE_VERSION).
+      (warnIf
+        (tuxedo-keyboard.version < "3.1.0")
```

Comment:
This should be `(builtins.compareVersions tuxedo-keyboard.version "3.1.0") < 0` otherwise this just does lexicographic comparison, which fails for some inputs:

```
nix-repl> "3.11.0" < "3.9.0"
true
```

Reply:
Duh, how could I have missed something like this. Well, today I learned; makes total sense that there is a correct way of doing this in nix. Should work correctly now.